pkg/connector/handlematrix.go

```diff
@@ -391,3 +398,112 @@ func (wa *WhatsAppClient) HandleMatrixMembership(ctx context.Context, msg *bridg
 
 	return true, nil
 }
+
+func (wa *WhatsAppClient) HandleMatrixRoomName(ctx context.Context, msg *bridgev2.MatrixRoomName) (bool, error) {
+	portalJID, err := waid.ParsePortalID(msg.Portal.ID)
+	if err != nil {
+		return false, err
+	}
+
+	if msg.Portal.RoomType == database.RoomTypeDM {
+		return false, fmt.Errorf("cannot set room name for DM")
+	}
+
+	err = wa.Client.SetGroupName(portalJID, msg.Content.Name)
+	if err != nil {
+		return false, err
+	}
+
```

Comment:
this will prevent the echo (topic and avatar need the same; for avatar you could set hash and mxc but not id to allow it to fix the avatar if whatsapp server modifies it):

```suggestion
	msg.Portal.Name = msg.Content.Name
	msg.Portal.NameSet = true
```